code/game/machinery/dna_scanner.dm

```diff
@@ -108,7 +108,7 @@
 			|| locate(/obj/machinery/computer/cloning, get_step(src, SOUTH)) \
 			|| locate(/obj/machinery/computer/cloning, get_step(src, EAST)) \
 			|| locate(/obj/machinery/computer/cloning, get_step(src, WEST)))
-			if(!occupant.suiciding && !(occupant.disabilities & NOCLONE))
+			if(!occupant.suiciding && !(occupant.disabilities & NOCLONE) && !occupant.hellbound)
```

Comment:
we really need some kind of ```can_revive()```/```can_clone()``` procs, this isn't your fault or responsibility though.